### PR TITLE
Issue #14 - use default values instead of throwing

### PR DESCRIPTION
**Pull request by mrshanahan**
Used MSBuild's defaults for AssemblyName & OutputType, but this resulted in a change in behavior: project files with valid XML but meaningless elements no longer throw an exception. I changed the unit tests to reflect that; now the extension just returns a package with no configurations.

This has the side effect of such a project returning the error message "File type is not supported" instead of throwing an exception when loaded by the console, but that behavior already occurs with projects that have complete MSBuild XML but no build configurations, so it seems to make sense.

### CODE REVIEW

**Comment by CharliePoole** [comment]
@ManfredLange You've looked a lot at this sort of thing. Would you do a review?

**Comment by ManfredLange** [comment]
@CharliePoole Sure, no problem.

**Comment by mrshanahan** [comment]
@ManfredLange Ahh sorry, somehow missed that you responded!
1. Great, thanks!
2. I repurposed an existing project file to check that a provided property was still used when the other was missing (i.e. that a missing AssemblyName didn't also trigger a default OutputType for some reason and vice versa). Maybe my reasoning is a bit weird there. The most thorough option would be to have 4 files for each combination of present/missing property, but that seemed like overkill.
3. :-)
4. That makes sense! There is public documentation on what elements are required and optional strictly in terms of the XML schema, but not for the properties inside a `<PropertyGroup>` element. Perhaps we should be checking that behavior for each XML element that is used. Something to keep in mind I suppose.

**Comment by mrshanahan** [comment]
Didn't realize this didn't close! @ManfredLange Is there anything I should do re: your last comment, or is this PR good as-is? I don't have write access, so I can't merge it myself.

**Comment by CharliePoole** [comment]
@ManfredLange I didn't see your comments either - I only looked at the email, which just had the first line! I'll review myself now as well.

File: src/tests/resources/csharp-missing-assembly-name.csproj [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <ProductVersion>8.0.30703</ProductVersion>
     <SchemaVersion>2.0</SchemaVersion>
     <ProjectGuid>{6BF38215-C9D9-418D-9D81-4DECBA679223}</ProjectGuid>
-    <OutputType>Library</OutputType>
+    <OutputType>Exe</OutputType>
```
Comment (CharliePoole) [inline src/tests/resources/csharp-missing-assembly-name.csproj]: Like Manfred, I don't understand why you made this change, could you explain please?

**Comment by CharliePoole** [comment]
Thanks for your work. I'm ready to approve and merge this pending the single question I left. The change doesn't hurt anything, but I don't understand the purpose, which never makes me happy.

File: src/tests/resources/csharp-missing-assembly-name.csproj (hunk 1 above)
Comment (mrshanahan) [reply src/tests/resources/csharp-missing-assembly-name.csproj]: The original use of this file was to make sure the extension threw when AssemblyName was missing. I repurposed it to 1) use a default AssemblyName but also to 2) use a non-default OutputType to ensure that both default values weren't being returned, as a bit of black box verification. I can certainly change it back if that's a bit obtuse, but it heads off a potential incorrect code path.

**Comment by CharliePoole** [comment]
Thanks again. Glad to have this fix in.

**Comment by mrshanahan** [comment]
Thank you both for taking the time to look at it!

**Pull request merged**